Common/Tests/Utilities.UI/UI/ReplWindowProxy.cs

```diff
@@ -103,43 +68,33 @@ >>> int('0b100', base=0)
 
         public bool InlinePrompts { get; set; }
 
-        public string SourceFileName { get; set; }
-
-        public string IPythonIntDocumentation { get; set; }
-
-        public string IntFirstMember { get; set; }
-
-        public string RawInput { get; set; }
-
-        public string Print42Output { get; set; }
-
-        public bool KeyboardInterruptHasTracebackHeader { get; set; }
-
         public bool EnableAttach { get; set; }
-
-        public string ImportError { get; set; }
     }
 
     internal sealed class ReplWindowProxy : IDisposable {
-        private readonly PythonVisualStudioApp _app;
-        private readonly ReplWindow _window;
+        private readonly VisualStudioApp _app;
+        private readonly ToolWindowPane _toolWindow;
+        private readonly IInteractiveWindow _window;
         private readonly ReplWindowProxySettings _settings;
         private readonly ReplWindowInfo _replWindowInfo;
         private readonly IEditorOperations _editorOperations;
 
+#if !DEV14_OR_LATER
         private Dictionary<ReplOptions, object> _restoreOptions;
+#endif
         private List<Action> _onDispose;
 
-        private static ConditionalWeakTable<ReplWindow, ReplWindowInfo> _replWindows =
-            new ConditionalWeakTable<ReplWindow, ReplWindowInfo>();
+        private static ConditionalWeakTable<ToolWindowPane, ReplWindowInfo> _replWindows =
+            new ConditionalWeakTable<ToolWindowPane, ReplWindowInfo>();
 
-        internal ReplWindowProxy(PythonVisualStudioApp app, ReplWindow window, ReplWindowProxySettings settings) {
+        internal ReplWindowProxy(VisualStudioApp app, IInteractiveWindow window, ToolWindowPane toolWindow, ReplWindowProxySettings settings) {
             Assert.IsNotNull(app, "app is required");
             Assert.IsNotNull(window, "window is required");
             _app = app;
             _window = window;
+            _toolWindow = toolWindow;
             _settings = settings;
-            _replWindowInfo = _replWindows.GetOrCreateValue(_window);
+            _replWindowInfo = _replWindows.GetOrCreateValue(toolWindow);

```

Comment:
That's barely even a nit... _toolWindow gets used elsewhere, but there's no need to use it here.